<launch>
  <!-- topic of your odometry such as VIO or LIO -->
  <arg name="odom_topic" value="/state_ukf/odom" />

  <!-- main algorithm params -->
  <include file="$(find plan_manage)/launch/racing.xml">

    <arg name="odometry_topic" value="$(arg odom_topic)"/>

    <!-- maximum velocity and acceleration the drone will reach -->
    <arg name="max_vel" value="15.0" />
    <arg name="max_acc" value="6.0" />

    <!-- global waypoints -->
    <!-- If flight_type is set to 2, the drone will travel these waypoints one by one -->
    <arg name="point_num" value="10" />

    <arg name="point0_x" value="0.0" />
    <arg name="point0_y" value="0.0" />
    <arg name="point0_z" value="1.0" />

    <arg name="point1_x" value="6.0" />
    <arg name="point1_y" value="4.0" />
    <arg name="point1_z" value="1.0" />

    <arg name="point2_x" value="12.0" />
    <arg name="point2_y" value="-4.0" />
    <arg name="point2_z" value="1.0" />

    <arg name="point3_x" value="18.0" />
    <arg name="point3_y" value="4.0" />
    <arg name="point3_z" value="1.0" />

    <arg name="point4_x" value="12.0" />
    <arg name="point4_y" value="10.0" />
    <arg name="point4_z" value="1.0" />

    <arg name="point5_x" value="0.0" />
    <arg name="point5_y" value="10.0" />
    <arg name="point5_z" value="1.0" />

    <arg name="point6_x" value="-6.0" />
    <arg name="point6_y" value="4.0" />
    <arg name="point6_z" value="1.0" />

    <arg name="point7_x" value="-12.0" />
    <arg name="point7_y" value="-4.0" />
    <arg name="point7_z" value="1.0" />

    <arg name="point8_x" value="-6.0" />
    <arg name="point8_y" value="-10.0" />
    <arg name="point8_z" value="1.0" />

    <arg name="point9_x" value="6.0" />
    <arg name="point9_y" value="-10.0" />
    <arg name="point9_z" value="1.0" />

  </include>

  <!-- data visualization -->
  <node name="data_plot" pkg="rqt_plot" type="rqt_plot" 
    args="/visualizer/speed_cmd /visualizer/real_speed /visualizer/total_thrust_cmd /visualizer/real_total_thrust">
  </node>

  <!-- trajectory server -->
  <node pkg="plan_manage" name="il_traj_server" type="il_traj_server" output="screen">
    <remap from="/position_cmd" to="planning/pos_cmd"/>

    <remap from="/odom_world" to="$(arg odom_topic)"/>
    <param name="traj_server/time_forward" value="1.5" type="double"/>
  </node>

  <node pkg="waypoint_generator" name="waypoint_generator" type="waypoint_generator" output="screen">
    <remap from="~odom" to="$(arg odom_topic)"/>        
    <remap from="~goal" to="/move_base_simple/goal"/>
    <remap from="~traj_start_trigger" to="/traj_start_trigger" />
    <param name="waypoint_type" value="manual-lonely-waypoint"/>    
  </node>

  <!-- use simulator -->
  <include file="$(find plan_manage)/launch/simulator.xml">
    <arg name="map_size_x_" value="40.0"/>
    <arg name="map_size_y_" value="20.0"/>
    <arg name="map_size_z_" value="5.0"/>
    <arg name="c_num" value="10"/>
    <arg name="p_num" value="80"/>

    <arg name="odometry_topic" value="$(arg odom_topic)" />
  </include>

</launch>
